pkg/kubefedctl/federate/federate.go

```diff
@@ -161,6 +161,13 @@ func NewCmdFederateResource(cmdOut io.Writer, config util.FedConfig) *cobra.Comm
 
 // Run is the implementation of the `federate resource` command.
 func (j *federateResource) Run(cmdOut io.Writer, config util.FedConfig) error {
+	var err error
+	if len(j.resourceNamespace) == 0 {
+		j.resourceNamespace, err = util.GetNamespace(j.HostClusterContext, j.Kubeconfig, config)
+	}
+	if err != nil {
```

Comment:
Please move this error handler to the block in which it is reported.

Reply:
Done